### DISCUSSION TITLE
nullable constraint

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#4789] [Language Ideas]
```
public T Null<T>() where T : nullable
{
    return null;
}

string someName1 = Null<string>();
int? someName2 = Null<int?>();
```

****
I was wrong.

**Comment by cheong00** [comment]
I think you should be putting null into nullable variables instead of corresponding nullable types.

Otherwise what's the point of creating nullables at all? Shouldn't making to magic value to every type be enough?

**Comment by huoyaoyuan** [comment]
`null` of reference types and `null` of nullable value types are totally different things in IL.

**Comment by CyrusNajmabadi** [comment]
I'm missing something. First, I don't think you need generics here. However, if you want a generic method here, it would be:

`T Default<T>() => default;`

This will give you `null` for both `string` and `int?`.  But, as you can see from the method body, this method is not necessary and you can just use `default` or `default(T)`
